packages/flutter_migrate/lib/src/compute.dart

```diff
@@ -0,0 +1,1014 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'base/common.dart';
+import 'base/file_system.dart';
+import 'base/logger.dart';
+import 'base/project.dart';
+import 'base/terminal.dart';
+import 'custom_merge.dart';
+import 'environment.dart';
+import 'flutter_project_metadata.dart';
+import 'result.dart';
+import 'utils.dart';
+
+// This defines files and directories that should be skipped regardless
+// of gitignore and config settings. Using `/` as a stand-in for path separator.
+const List<String> _skippedFiles = <String>[
+  'ios/Runner.xcodeproj/project.pbxproj', // Xcode managed configs that may not merge cleanly.
+  'README.md', // changes to this shouldn't be overwritten since is is user owned.
+];
+
+const List<String> _skippedDirectories = <String>[
+  '.dart_tool', // ignore the .dart_tool generated dir
+  '.git', // ignore the git metadata
+  'lib', // Files here are always user owned and we don't want to overwrite their apps.
+  'test', // Files here are typically user owned and flutter-side changes are not relevant.
+  'assets', // Common directory for user assets.
+  'build' // Build artifacts
+];
+
+bool _skipped(String localPath, FileSystem fileSystem,
+    {Set<String>? blacklistPrefixes}) {
+  for (final String path in _skippedFiles) {
+    if (path.replaceAll('/', fileSystem.path.separator) == localPath) {
+      return true;
+    }
+  }
+  if (_skippedFiles.contains(localPath)) {
+    return true;
+  }
+  for (final String dir in _skippedDirectories) {
+    if (localPath.startsWith(
+        '${dir.replaceAll('/', fileSystem.path.separator)}${fileSystem.path.separator}')) {
+      return true;
+    }
+  }
+  if (blacklistPrefixes != null) {
+    for (final String prefix in blacklistPrefixes) {
+      if (localPath.startsWith('$prefix${fileSystem.path.separator}')) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+const List<String> _skippedMergeFileExt = <String>[
+  // Don't merge image files
+  '.png',
+  '.jpg',
+  '.jpeg',
+  '.gif',
+  '.bmp',
+  '.svg',
+  // Don't merge compiled artifacts and executables
+  '.jar',
+  '.so',
+  '.exe',
```

Comment:
If this is something that's actually going to happen, shouldn't we be using file type checks of some kind instead? This will only catch Windows executables, not macOS or Linux executables.

Reply:
This is not meant to comprehensively catch all executables. This particular list is to catch executables that are included in the vanilla flutter project, which this list sufficiently covers. We don't need the overhead and boilerplate of formally checking executable bits of permissions. I can document this.

Reply:
Do we have `exe` files in Flutter projects by default?